content/docs/user-guide/how-to/update-tracked-files.md

```diff
@@ -1,76 +1,71 @@
+[replacing](#replacing-file) it with a new one (same file name).
 
-> For an example of the cache corruption problem see
-> [issue #599](https://github.com/iterative/dvc/issues/599) in our GitHub
-> repository.
+When the `cache.type` config option is set to `symlink` or `hardlink` (not the
+default, see `dvc config cache` for more info.), updating tracked files has to
+be carried out with caution, to avoid data corruption. This is due to the way in
+which DVC handles linking data files between the <abbr>cache</abbr> and the
+<abbr>workspace</abbr> (refer to
+[Large Dataset Optimization](/doc/user-guide/large-dataset-optimization) for
+details).
 
-Assume `train.tsv` is tracked by DVC and you want to update it. Here updating
-may mean either replacing `train.tsv` with a new file having the same name or
-editing the content of the file.
+> For an example of the cache corruption problem see
+> [issue #599](https://github.com/iterative/dvc/issues/599) in our GitHub repo.
 
-If you run `dvc repro` there is no need to manage generated (output) files
-manually. DVC removes them for you before executing the stage that generates
+If you use `dvc.yaml` files and `dvc repro`, there is no need to manage stage
+<abbr>outputs</abbr> manually. DVC removes them for you before regenerating
 them.
 
-If you use DVC to track a file that is generated during your pipeline (e.g. some
-intermediate result or a final model file i.e. `model.pkl`) and you don't use
-`dvc run` and `dvc repro` to manage your pipeline, use the procedure below (run
-`dvc unprotect` or `dvc remove`) to unlink it from DVC cache prior to the
-execution of the script that modifies it.
-
-See also `dvc unprotect` and `dvc config cache` to learn more about protecting
-your data files.
+Otherwise (the data was tracked with `dvc add`), use one of the procedures below
+to unlink the data from the cache prior to updating it. We'll be working with a
+`train.tsv` file:
 
-## Replacing file
-
-If you want to replace the file, you can take the following steps.
+## Modifying content
 
-First, un-track the file. This will remove `train.tsv` from the workspace:
+"Unlink" the file with `dvc unprotect`. This will make `train.tsv` safe to edit:
 
 ```dvc
-$ dvc remove train.tsv.dvc
+$ dvc unprotect train.tsv
 ```
 
-Next, replace the file with new content:
+Then edit the content of the file:
 
 ```dvc
-$ echo new > train.tsv
+$ echo "new data item" >> train.tsv
 ```
 
-And start tracking it again:
+Add the new version of the file back with DVC:
 
 ```dvc
 $ dvc add train.tsv
-$ git add train.tsv.dvc .gitignore
-$ git commit -m "new train data"
+$ git add train.tsv.dvc
+$ git commit -m "modify train data"
 ```
 
-## Modifying content
+## Replacing files
 
-"Unlink" the file with `dvc unprotect`. This will make `train.tsv` safe to edit:
+If you want to replace the file, you can take the following steps.
+
+First, [un-track](/doc/user-guide/how-to/untrack-data) the file with
```

Comment:
Also, it now links to **How to Un-track Data** how-to from the **Replacing files** section. There's definitely some overlap there (around `remove`).

We could consider merging this section and that doc, not sure.